### PR TITLE
fix: check that frozen.build exists before trying to access it

### PR DESCRIPTION
**Pull request by conorh**
If there is a compilation where frozen.build is not generated for some reason then the check here would fail and throw an exception:
```
   Could not freeze FILE_NAME: Cannot read property 'hash' of undefined
```
Added check to see if the build exists, and then fall through to regenerate if it does not.

### CODE REVIEW

**Comment by vlasenko** [comment]
@mzgoddard Can we have this merged?

**Comment by vlasenko** [comment]
@mzgoddard Could you please, please review this PR and merge it? This problem is really a showstopper for us.

**Comment by eddieferrer** [comment]
Any update on merging this in?

**Comment by bovas85** [comment]
bump

**Comment by elevatebart** [comment]
Can anyone look at the travis tests, see why they are not passing?

**Comment by ansballard** [comment]
So the tests aren't passing because `npm run lint` returns errors (calls `prettier --list-different ...`). Running  `npm run lint:fix` should get travis to pass, as confirmed [on a branch I made from this one](https://github.com/mzgoddard/hard-source-webpack-plugin/compare/master...ansballard:fix-cache?expand=1). I actually went back a year or so in the travis build history and not a single run has succeeded, so we could merge this as is, _especially_ since it's been tested via use of [hard-source-webpack-plugin-fixed-hashbug](https://www.npmjs.com/package/hard-source-webpack-plugin-fixed-hashbug).

TL;DR this looks perfectly safe to merge, with or without fixing the liner errors that are already in master anyway.

**Comment by dazjean** [comment]
m

**Review by rlods — APPROVED** [review]
